src/daos/lib/IPFSLibraryDAO.js

```diff
@@ -0,0 +1,83 @@
+import web3 from 'web3'
+import AbstractContractDAO from './AbstractContractDAO'
+import HashSetEvent from "../../models/app/HashSetEvent";
+import {ipfsHashToBytes32} from "../../utils";
+
+export default class IPFSLibraryDAO extends AbstractContractDAO {
+  constructor ({ address, history, abi }) {
+    super({ address, history, abi })
+  }
+
+  async connect (web3, options) {
+    if (this.isConnected) {
+      this.disconnect()
+    }
+    // eslint-disable-next-line no-console
+    console.log(`[${this.constructor.name}] Connect`)
+    this.contract = new web3.eth.Contract(this.abi.abi, this.address, options)
+    this.history = this.history != null // nil check
+      ? new web3.eth.Contract(this.abi.abi, this.history, options)
+      : this.contract
+
+    this.hashSetEmitter = this.history.events.HashSet({})
+      .on('data', this.handleHashSetData.bind(this))
+      .on('error', this.handleError.bind(this))
+
+    return this.token
+  }
+
+  disconnect () {
+    if (this.isConnected) {
+      this.contract = null
+      this.history = null
+      this.hashSetEmitter.removeAllListeners()
+      this.hashSetEmitter = null
+    }
+  }
+
+  get isConnected () {
+    return this.contract != null // nil check
+  }
+
+  async getHash (from, itemName) {
+    const itemNameInBytes32 = web3.utils.asciiToHex(itemName)
```

Comment:
https://github.com/ChronoBank/Labor-X/blob/develop/src/utils/bytes32.js#L7
Does this action make exactly the same thing?

Reply:
imho these algorythms differs
https://github.com/cryptocoinjs/base-x/blob/master/index.js#L51
https://github.com/ethereum/web3.js/blob/v1.0.0-beta.33/packages/web3-utils/src/index.js

found here instructions how to convert string to bytes32 and back 
https://ethereum.stackexchange.com/questions/23058/web3-return-bytes32-string

Reply:
Everything fine